### PR TITLE
Make project access form discoverable

### PR DESCRIPTION
**Pull request by debsmita1**
**JIRA story:**
https://issues.redhat.com/browse/ODC-6636

**Solution description:**
- Added `Sharing` section on the Add page
- Clicking on it takes the user to the project access form
- the card is shown to users who have the privilege to create & list role bindings

**GIF:**
pa

**Test coverage:**
test1

/kind feature

### CODE REVIEW

File: frontend/packages/dev-console/console-extensions.json [hunk 1]
```diff
@@ -650,5 +664,13 @@
     "flags": {
       "required": ["ALLOW_SERVICE_BINDING"]
     }
+  },
+  {
+    "type": "console.page/route",
+    "properties": {
+      "exact": true,
+      "path": ["/project-access/ns/:ns"],
```
Comment (vikram-raj) [inline frontend/packages/dev-console/console-extensions.json:672]: Can we also add `/project-access/all-namespaces` path. and handle the `all-namespaces` use case same as we do for other dev console pages?

Currently `/project-access/all-namespaces` shows 404 page.

File: frontend/packages/dev-console/src/actions/add-resources.tsx [hunk 2]
```diff
@@ -158,6 +159,16 @@ export const AddActions: { [name: string]: ActionFactory } = {
     path,
     disabled: accessReviewDisabled,
   }),
+  ProjectAccess: (namespace, application, contextSource, path, accessReviewDisabled) => ({
```
Comment (vikram-raj) [inline frontend/packages/dev-console/src/actions/add-resources.tsx:162]: Do we need `Sharing` option in the topology context menu? I do not see this AC in the story.

File: frontend/packages/dev-console/console-extensions.json [hunk 3]
```diff
@@ -186,6 +186,21 @@
...
+      "id": "project-access",
+      "href": "/project-access/ns/:namespace",
+      "label": "%devconsole~Sharing%",
+      "description": "%devconsole~Project access allows you to add or remove a user's access to the project.%",
+      "icon": { "$codeRef": "icons.shareProjectIconElement" },
+      "accessReview": [
+        { "group": "rbac.authorization.k8s.io", "resource": "rolebindings", "verb": "create" },
+        { "group": "rbac.authorization.k8s.io", "resource": "rolebindings", "verb": "list" }
+      ],
+      "insertBefore": "local-machine"
```
Comment (invincibleJai) [inline frontend/packages/dev-console/console-extensions.json:201]: I don't see `insertBefore` in the definition for type `dev-console.add/action` here https://github.com/openshift/console/blob/master/frontend/packages/console-dynamic-plugin-sdk/src/extensions/add-actions.ts#L5-L23

**Comment by kjack-redhat** [comment]
looks good to me

File: frontend/packages/dev-console/console-extensions.json [hunk 4]
```diff
@@ -186,6 +186,30 @@
...
+    }
+  },
+  {
+    "type": "dev-console.add/action",
+    "properties": {
+      "id": "project-access",
+      "groupId": "sharing",
+      "href": "/project-access/ns/:namespace",
+      "label": "%devconsole~Sharing%",
+      "description": "%devconsole~Project access allows you to add or remove a user's access to the project.%",
```
Comment (vikram-raj) [inline frontend/packages/dev-console/console-extensions.json:205]: nit: for the consistency can we remove the full stop (.) As we do not have it for any descriptions of the add action on add page.
```suggestion
      "description": "%devconsole~Project access allows you to add or remove a user's access to the project%",
```

**Review by vikram-raj — COMMENTED** [review]
/lgtm

**Comment by debsmita1** [comment]
Propagating docs and px acks from the Epic
/label docs-approved
/label px-approved

**Pull request merged**